lib/Basics/Thread.cpp

```diff
@@ -182,62 +178,37 @@ void Thread::beginShutdown() {
     _state.compare_exchange_strong(state, ThreadState::STOPPED);
   }
 
-  while (state != ThreadState::STOPPING && state != ThreadState::STOPPED &&
-         state != ThreadState::DETACHED) {
+  while (state != ThreadState::STOPPING && state != ThreadState::STOPPED) {
     _state.compare_exchange_strong(state, ThreadState::STOPPING);
```

Comment:
Could we use the weak variant in the loop?

Reply:
Yes, since this is inside a loop the weak variant would be sufficient.